### PR TITLE
Add Levenshtein distance function

### PR DESCRIPTION
**Pull request by Crayder**
Adds `strdistance`.

### CODE REVIEW

**Comment by Crayder** [comment]
Tested with the following code;

```pawn
#include <a_samp>
#include <strlib>

main()
{
	print("Regular Strings");
	
	printf("\ttest  -> Test   (0, 1) : %i, %i" 		, strdistance("test", "Test", true)		, strdistance("test", "Test")		);
	printf("\ttest  -> tesTT  (1, 2) : %i, %i" 		, strdistance("test", "tesTT", true)	, strdistance("test", "tesTT")		);
	printf("\ttest2 -> TEst   (1, 3) : %i, %i" 		, strdistance("test2", "TEst", true)	, strdistance("test2", "TEst")		);
	printf("\ttest  -> teSTtt (2, 3) : %i, %i" 		, strdistance("test", "teSTtt", true)	, strdistance("test", "teSTtt")		);
	                                                
	print("Packed Strings");                        
	                                                
	printf("\ttest  -> Test   (0, 1) : %i, %i" 		, strdistance(!"test", !"Test", true)	, strdistance(!"test", !"Test") 	);
	printf("\ttest  -> tesTT  (1, 2) : %i, %i"  	, strdistance(!"test", !"tesTT", true)	, strdistance(!"test", !"tesTT")	);
	printf("\ttest2 -> TEst   (1, 3) : %i, %i"  	, strdistance(!"test2", !"TEst", true)	, strdistance(!"test2", !"TEst")	);
	printf("\ttest  -> teSTtt (2, 3) : %i, %i" 		, strdistance(!"test", !"teSTtt", true)	, strdistance(!"test", !"teSTtt")	);
	
	print("strpack Strings");
	
	printf("\ttest  -> Test   (0, 1) : %i, %i" 		, strdistance(ret_strpack("test"), ret_strpack("Test"), true)	, strdistance(ret_strpack("test"), ret_strpack("Test"))  	);
	printf("\ttest  -> tesTT  (1, 2) : %i, %i"  	, strdistance(ret_strpack("test"), ret_strpack("tesTT"), true)	, strdistance(ret_strpack("test"), ret_strpack("tesTT")) 	);
	printf("\ttest2 -> TEst   (1, 3) : %i, %i"  	, strdistance(ret_strpack("test2"), ret_strpack("TEst"), true)	, strdistance(ret_strpack("test2"), ret_strpack("TEst")) 	);
	printf("\ttest  -> teSTtt (2, 3) : %i, %i" 		, strdistance(ret_strpack("test"), ret_strpack("teSTtt"), true) , strdistance(ret_strpack("test"), ret_strpack("teSTtt"))	);
}
```

Results;

```
Regular Strings
        test  -> Test   (0, 1) : 0, 1
        test  -> tesTT  (1, 2) : 1, 2
        test2 -> TEst   (1, 3) : 1, 3
        test  -> teSTtt (2, 3) : 2, 3
Packed Strings
        test  -> Test   (0, 1) : 0, 1
        test  -> tesTT  (1, 2) : 1, 2
        test2 -> TEst   (1, 3) : 1, 3
        test  -> teSTtt (2, 3) : 2, 3
strpack Strings
        test  -> Test   (0, 1) : 0, 1
        test  -> tesTT  (1, 2) : 1, 2
        test2 -> TEst   (1, 3) : 1, 3
        test  -> teSTtt (2, 3) : 2, 3
```

**Comment by Crayder** [comment]
Example use cases (both quite recent actually):

- https://github.com/maddinat0r/sscanf/issues/25
- https://forum.sa-mp.com/showthread.php?p=4092313#post4092313

File: strlib.inc [hunk 1]
```diff
@@ -523,6 +537,68 @@ stock bool:isequal(const str1[], const str2[], bool:ignorecase = false) {
     return !strcmp(str1, str2, ignorecase);
 }
 
+stock strdistance(const str1[], const str2[], bool:ignorecase = false) {
+	// If they are equal
+	if(isequal(str1, str2, ignorecase))
+		return 0;
+
+	// Data
+	static d[128][128];
```
Comment (Southclaws) [inline strlib.inc]: Perhaps just name the variable `data` and you won't need the comment explaining what the variable is.

Comment (Crayder) [reply strlib.inc]: I mean, yeh.

**Comment by oscar-broman** [comment]
👍🏻

**Pull request merged**